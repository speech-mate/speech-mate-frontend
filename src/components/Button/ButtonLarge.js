import React from "react";
import styled from "styled-components";
import propTypes from "prop-types";

function ButtonLarge({ text, onClick, node }) {
  return (
    <LargeButton onClick={onClick} data-node={node}>
      {text}
    </LargeButton>
  );
}

const LargeButton = styled.button`
  width: 90%;
  height: 60px;
  border: none;
  border-radius: 10px;
  background-color: var(--maximum-blue-purple-color);
  color: white;
  font-size: 20px;
`;

ButtonLarge.propTypes = {
  text: propTypes.string.isRequired,
  onClick: propTypes.func.isRequired,
  node: propTypes.node,
};

export default ButtonLarge;
