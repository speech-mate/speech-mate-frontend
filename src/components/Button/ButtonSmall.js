import React from "react";

function ButtonSmall() {
  return <div>ButtonSmall</div>;
}

export default ButtonSmall;
