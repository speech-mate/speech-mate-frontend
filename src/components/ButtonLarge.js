import React from "react";
import styled from "styled-components";
import propTypes from "prop-types";

function ButtonLarge({ text, onClick }) {
  return <LargeButton onClick={onClick}>{text}</LargeButton>;
}

const LargeButton = styled.button`
  width: 90%;
  height: 65px;
  border: none;
  border-radius: 10px;
  background-color: var(--maximum-blue-purple-color);
  color: white;
  font-size: 20px;
`;

ButtonLarge.propTypes = {
  text: propTypes.string.isRequired,
  onClick: propTypes.func.isRequired,
};

export default ButtonLarge;
