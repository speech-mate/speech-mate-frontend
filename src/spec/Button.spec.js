import React from "react";
import { screen, render, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import ButtonLarge from "../components/Button/ButtonLarge";
import ButtonSmall from "../components/Button/ButtonSmall";

afterEach(cleanup);

const buttonLarge = (
  <ButtonLarge type="button" onClick={() => console.log("click")} text="test" />
);
const buttonSmall = (
  <ButtonSmall type="button" onClick={() => console.log("click")} text="test" />
);

describe("<ButtonLarge />", () => {
  it("matches snapshot", () => {
    const { container } = render(buttonLarge);

    expect(container).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    const { getByText } = render(buttonLarge);

    expect(getByText("test")).toBeInTheDocument();
  });

  it("should occur an event", () => {
    jest.spyOn(global.console, "log");

    render(buttonLarge);

    const button = screen.getByText("test");

    userEvent.click(button);

    expect(console.log).toHaveBeenCalledWith("click");
  });
});

describe("<ButtonSmall />", () => {
  it("matches snapshot", () => {
    const { container } = render(buttonSmall);

    expect(container).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    const { getByText } = render(buttonSmall);

    expect(getByText("test")).toBeInTheDocument();
  });

  it("should occur an event", () => {
    jest.spyOn(global.console, "log");

    render(buttonSmall);

    const button = screen.getByText("test");

    userEvent.click(button);

    expect(console.log).toHaveBeenCalledWith("click");
  });
});
