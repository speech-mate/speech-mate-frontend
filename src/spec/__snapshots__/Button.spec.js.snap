// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonLarge /> matches snapshot 1`] = `
<div>
  <button
    class="sc-bdvvtL cwdsrI"
  >
    test
  </button>
</div>
`;

exports[`<ButtonSmall /> matches snapshot 1`] = `
<div>
  <button
    class="sc-gsDKAQ jiAfAQ"
  >
    test
  </button>
</div>
`;
